cheatsheets_draft/Webhook_Security_Guidelines_CheatSheet.md

```diff
@@ -0,0 +1,169 @@
+
+At high level, the following requirements outline the security mechanisms in various ways. These high level requirements allow engineers and developers to properly implement security controls. 
+
+
+- **Confidentiality: A passive attacker must not be able to read Webhook Notifications, as they may contain confidential data.**
+- **Integrity: An active attacker must not be able to alter Webhook Notifications, as this may result in unexpected behaviour affecting your infrastructure.**
+- **Authenticity: An active attacker must not be able to forge Webhook Notifications as this may result in your receive code taking unwanted action.**
+
+
+## Contents
+
+Controls for Securing Webhooks
+
+- Securing Webhook URLs
+-	Enable SSL/TLS
+-	Basic Authentication
+- Mutual TLS Authentication
+-	Oath Authorization
+-	Crude Authorized Access
+-	Message Protection and Secrecy
+-	Cross Site Request Forgery
+-	Rotation and Retry
+-	Pass Secrets in Request header
+-	Fail Safe and Secure
+-	Replay Protections
+-	Payload best practices
+-	HTTP Methods best practices
+
+
+## Control 1 - Securing Webhook URLs
+
+Secure URLs - Keep URLs out of source code, keep urls out of logs. Do not print Webhook URLs in logs, Do not store write URLs or hard code it in source. Webhook URLs should be securely stored in Vault. 
+
+Abuse Case: 
+- Find a Webhook URL that is in code and try to post a simple message to it.
+
+
+## Control 2 - Transport Security 
+
+Enable SSL/TLS - The first step you should take to secure web application is to ensure that you are using HTTPS for web application's end point. Data transmitted over the Internet should always be encrypted. Your endpoint URL should support HTTPS, and you should add that secure URL to your webhook settings. Avoid connecting to an HTTPS URL with a self-signed certificate and weak Ciphers. Server must be correctly configured to support HTTPS with a valid server certificate. 
+
+Abuse Case: 
+- Use an existing webhook and try to make a request to it using http://. 
+- Alternative abuse case is to negotiate weak SSL/TLS when making a webhook request. 
+- Alternative abuse case is to look at the certificate and see if it is self-signed. If it does not follow a strong chain of trust it should not be trusted for sensitive data or production traffic unless other mechanisms are in place
+
+
+## Control 3 - Basic Authentication 
```

Comment:
Ack, frankly I still use Basic Authentication for webhooks i have developed :) However, I have come across some companies removing Basic Auth. 

Below one is from Hubspot. 
In an effort to enforce the highest security standards,  basic authentication is being removed from webhooks within workflows. There’s a new, more secure way to help you verify your webhook requests coming from Hubspot.
https://developers.hubspot.com/changelog/sunset-basic-authentication-with-webhooks-will-be-sunset-april-8th

Would need help here to rephrase please :)

Reply:
Not everyone is capable to set signatures and their structure would require basic auth. Basic Auth in proper implementations can serve the needs of a system. It could use tokens as passwords, which make things much similar to normal requests. Let me know the points you'd want to tackle in a bulleted list, and I'd love to word this and help with it.

Reply:
Make it a sub-point in a bigger point authentication